gtsfm/utils/io.py

```diff
@@ -153,13 +169,15 @@ def write_cameras(gtsfm_data: GtsfmData, images: List[Image], save_dir: str) ->
             f.write(f"{i} {camera_model} {image_width} {image_height} {fx} {u0} {v0} {k1} {k2}\n")
 
 
-def write_images(gtsfm_data: GtsfmData, save_dir: str) -> None:
+def write_images(gtsfm_data: GtsfmData, means: np.ndarray, align_rot: np.ndarray, save_dir: str) -> None:
```

Comment:
I dont think we should have a transform in the write function. We should have a different function to transform into a new `GtsfmData` object.

Reply:
yup. I'll create another function in `ellipsoid.py` which will perform the camera frustum transformations. That way, `write_images()` inside `io.py` can just call that before writing the quaternions and translations to file